src/middlewares/errorHandler.js

```diff
@@ -0,0 +1,22 @@
+import Debug from 'debug';
+import { config } from 'dotenv';
+
+config();
+const debug = Debug('dev');
+export default (err, request, response, next) => {
+  const isProduction = process.env.NODE_ENV === 'production';
+
+  let errorMessage = {};
+  if (!isProduction) {
+    debug(err.stack);
+    errorMessage = err;
+  }
+  response.status(err.status || 500);
+  response.json({
+    errors: {
+      message: err.message,
+      error: errorMessage
+    }
+  });
+  next();
```

Comment:
The way express recognises an error handler is when it has 4 parameters `err` `req` `res` `next`. Since the errorhandler is the last line in the hierarchy there really is no need to call it, but I have to declare `next` as one of the parameters so I called it so as for it to pass the lint rules.

Reply:
This makes sense, you can add this logic to check if response headers have already been sent

```javascript
    if (res.headersSent) {
      return next(err);
    }
```

It uses next and handles an edge case. It's also inside the express documentation

Reply:
```javascript
    if (res.headersSent) {
      return next(err);
    }
```
This code is unreachable once i return the response